content/knowledge-codemagic/build-inputs.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Build inputs
+description: Specify optional inputs that are passed to the workflow when starting a build
+weight: 6
+---
+
+## Overview
+
+Build inputs are customizable parameters you can define within your workflow to make it more adaptable to different scenarios. With build inputs, you can create a single workflow and run it with different configurations by providing the values for inputs when starting a build for the workflow. For example, you can use build inputs to determine whether to build the workflow for test or release purposes or which app flavor to build. This eliminates the need to create multiple similar workflows with specific hardcoded values, making the workflow more reusable and dynamic.
+
+Inputs are workflow-specific and are defined in `codemagic.yaml` under the `inputs` mapping (see the [example](#minimal-example) below). The started workflow receives specified input values in the `inputs` context, i.e. `${{ inputs.inputId }}` is replaced with the value passed to input with identifier `inputId`.
+
+### Minimal example
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  greetings:
+    inputs:
+      name:
+        description: Who is greeted?
+        required: false
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting Builds with Inputs
+
+To successfully start a build, all required inputs must be specified. For optional inputs (those with required: false), the provided default value will be used if no value is specified when the build is started.
+
+{{<notebox>}}
+**Note**: Builds will fail if invalid values are provided for inputs (strings for numbers inputs, undefined choice options, etc.) or values are missing for required inputs.
+{{</notebox>}}
+
+### Starting Builds Manually via Codemagic UI
```

Comment:
```suggestion
### Starting builds manually via Codemagic UI
```

Reply:
Resolved in [ef40ff7](https://github.com/codemagic-ci-cd/codemagic-docs/pull/2537/commits/ef40ff791410c359c31f08359bb843fd96004fa8).